sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -613,7 +613,13 @@ class ParquetFilters(
     value == null || (nameToParquetField(name).fieldType match {
       case ParquetBooleanType => value.isInstanceOf[JBoolean]
       case ParquetIntegerType if value.isInstanceOf[Period] => true
-      case ParquetByteType | ParquetShortType | ParquetIntegerType => value.isInstanceOf[Number]
+      case ParquetByteType | ParquetShortType | ParquetIntegerType => value match {
+        // Byte/Short/Int are all stored as INT32 in Parquet so filters are built using type Int.
+        // We don't create a filter if the value would overflow.
+        case _: JByte | _: JShort | _: Integer => true
+        case v: JLong => v.longValue() >= Int.MinValue && v.longValue() <= Int.MaxValue
```

Comment:
For simply, how about forbid the Long value direclty ?

Reply:
There are some use cases where this can be useful, assuming the Parquet file contains type INT32:
- The user specifies a read schema using `schema("col LONG")`.
- The column in the table schema has type LONG.
I don't know whether the latter can happen today but it will be possible in the near future for Delta tables as I'm looking into supporting type widening.

We could skip creating row group filters in that case but the logic is simple enough and it's going to be beneficial in the cases above.

Reply:
The plan would be to support this in the Delta version that will build on top of the next Spark version, so it would be good to have it here already.

Reply:
If Spark already supports long type for a few releases, we can't drop it now or we have perf regressions. I'm +1 for the change here.